### ISSUE TITLE
Explicit paging

### ISSUE THREAD

**Issue by geneorama** [Chicago/RSocrata#31]
Currently paging is being done implicitly, we want to start doing it explicitly.  This should help with a number of issues, including memory management, which becomes a problem when requesting many (relatively) small pages of 1k rows each on large data sets.

Externally the function will need a "number of rows per page" function argument.
Internally the function will need to track the total row count for the data set, and put each page essentially into a pre-allocated slot determined by the dimensions of the table being returned.

Current (approx):

``` R
while (nrow(page) > 0) { 
        query <- validUrl + current_row_count_info
        response <- getResponse(query)
        page <- getContentAsDataFrame(response)
        # accumulate
        result <- rbind(result, page) 
}    
```

Proposed (approx)

``` R
total_rows <- query_to_get_row_count()
total_columns <- query_to_get_column_count()
result <- structure(NA, class = 'data_frame', attr('dim'=c(tot_rows, tot_cols)))
n_pages <- trunc(total_rows / rowlimit) + 1
for(i in 1: n_pages) { 
        query <- validUrl + exact_number_of_rows_to_get
        response <- getResponse(query)
        page <- getContentAsDataFrame(response)
        result[((i-1)*rowlimit+1):min(tot_rows, (i*rowlimit))] <- page
}
```

One of the errors that I've been getting when requesting large data sets is an "out of memory" error, which probably happens during the paging.  Currently there is a function in `data.table` called `rbindlist` that is specifically designed to accumulate pages, and is far more memory efficient than `data.frame`.  I used this in the FI model and I didn't have any out of memory errors. I don't know if that's what solved the problem, but it's something to keep in mind for the future.

**Comment by geneorama** [comment]
PS: The implicit row paging was getting 1k rows per request, but Socrata has increased the limit to 50k.

**Comment by nicklucius** [comment]
@geneorama @tomschenkjr - If we set paging at 50k, I am thinking it would render this issue moot. Any thoughts?

**Comment by geneorama** [comment]
@nicklucius setting paging at 50k would be explicit paging in my mind.  Currently the page size is implicit in the call.  We don't request 1k rows, we just get 1k rows because there is no `limit` argument. 

Not sure if that's what you're asking.  Maybe the short answer is "yes".

**Comment by tomschenkjr** [comment]
Sorry, this issue is closed as of 2845ff9 and I didn't catch it.

Earlier forms of the code had an implicit paging. [Here](https://github.com/Chicago/RSocrata/blob/b59e80729e4b00d5c87e0bde23497fbf00bebee5/R/RSocrata.R#L176-L181) is the paging code at v1.2.0 that did not explicitly set the paging limit.

**Comment by tomschenkjr** [comment]
That said, increasing the limit to 50k is a good thing.
